docs/api.md

```diff
@@ -6,180 +6,192 @@
+  * [jsHandle.jsonValue()](#jshandlejsonvalue)
+- [class: ElementHandle](#class-elementhandle)
+  * [elementHandle.asElement()](#elementhandleaselement)
+  * [elementHandle.boundingBox()](#elementhandleboundingbox)
+  * [elementHandle.click([options])](#elementhandleclickoptions)
+  * [elementHandle.dispose()](#elementhandledispose)
+  * [elementHandle.executionContext()](#elementhandleexecutioncontext)
+  * [elementHandle.focus()](#elementhandlefocus)
+  * [elementHandle.getProperties()](#elementhandlegetproperties)
+  * [elementHandle.getProperty(propertyName)](#elementhandlegetpropertypropertyname)
+  * [elementHandle.hover()](#elementhandlehover)
+  * [elementHandle.jsonValue()](#elementhandlejsonvalue)
+  * [elementHandle.press(key[, options])](#elementhandlepresskey-options)
+  * [elementHandle.screenshot([options])](#elementhandlescreenshotoptions)
+  * [elementHandle.tap()](#elementhandletap)
+  * [elementHandle.toString()](#elementhandletostring)
+  * [elementHandle.type(text[, options])](#elementhandletypetext-options)
+  * [elementHandle.uploadFile(...filePaths)](#elementhandleuploadfilefilepaths)
+- [class: Request](#class-request)
+  * [request.abort()](#requestabort)
+  * [request.continue([overrides])](#requestcontinueoverrides)
+  * [request.headers](#requestheaders)
+  * [request.method](#requestmethod)
+  * [request.postData](#requestpostdata)
+  * [request.resourceType](#requestresourcetype)
+  * [request.response()](#requestresponse)
+  * [request.url](#requesturl)
+- [class: Response](#class-response)
+  * [response.buffer()](#responsebuffer)
+  * [response.headers](#responseheaders)
+  * [response.json()](#responsejson)
+  * [response.ok](#responseok)
+  * [response.request()](#responserequest)
+  * [response.status](#responsestatus)
+  * [response.text()](#responsetext)
+  * [response.url](#responseurl)
 
 <!-- tocstop -->
 
-## Puppeteer
+### Overview
 
 Puppeteer is a Node library which provides a high-level API to control Chromium over the DevTools Protocol.
 
+Puppeteer API is hierarchical and mirrors browser objects. On the following diagram, faded entities are not currently represented in Puppeteer.
+
+![puppeteer overview](https://user-images.githubusercontent.com/746130/31592143-089f6f9a-b1db-11e7-9a20-16b7fc754fa1.png)
+
+- [`Puppeteer`](#class-puppeteer) communicates with browser using [devtools protocol](https://chromedevtools.github.io/devtools-protocol/).
+- [`Browser`](#class-browser) instance owns multiple pages.
+- [`Page`](#class-page) has at least one frame: main frame. There might be other frames created by [iframe](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe) or [frame](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/frame) tags.
+- [`Frame`](#class-frame) has at least one execution context - default execution context - where frame's javascript is executed. Frame might have additional execution contexts associated with [extensions](https://developer.chrome.com/extensions).
```

Comment:
JavaScript

Reply:
I'm *this close* to adding a lint check. Done.